src/reader.rs

```diff
@@ -3329,6 +3377,69 @@ impl<'a> Reader<'a> {
             }
         }
     }
+
+    fn _backward_token(&mut self) -> Option<(usize, EditableLineTag, usize)> {
```

Comment:
why the underscore?

Reply:
the function felt a bit specific to `ForwardToken` and `KillToken`. 

I'll remove the underscore

I'm also open to removing the functions and inlining the code in the four functions but AS-IS helps to make the differences between `ForwardToken` and `KillToken` more obvious